go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -1006,11 +1006,17 @@ func (h *defaultNetworkControllerEventHandler) UpdateResource(oldObj, newObj int
 			// Also check if node subnet changed, so static routes are properly set
 			// Also check if the node is used to be a hybrid overlay node
 			syncZoneIC = syncZoneIC || h.oc.isLocalZoneNode(oldNode) || nodeSubnetChange || zoneClusterChanged ||
-				switchToOvnNode || nodeEncapIPsChanged
+				switchToOvnNode || nodeEncapIPsChanged || primaryAddrChanged(oldNode, newNode)
```

Comment:
`primaryAddrChanged` checks `node.Status` to find primary IP, while you seem to be using `node-primary-ifaddr` annotation in the PR, please help me understand how you use both

Reply:
thanks for the pointer, confusing naming :)

will add an equivalent for the `node-primary-ifaddr`